### PR TITLE
CASSANDRA-19791: Remove other uses of Apache Commons lang for hashcode, equality and r…

### PR DESCRIPTION
**Pull request by jberragan**
…andom string generation

### CODE REVIEW

**Review by frankgh — APPROVED** [review]
+1 Looks good to me

File: cassandra-analytics-common/src/main/java/org/apache/cassandra/spark/utils/RandomUtils.java [hunk 1]
```diff
@@ -63,6 +64,24 @@ public static int randomPositiveInt(int bound)
         return RANDOM.nextInt(bound - 1) + 1;
     }
 
+    public static int nextInt(final int startInclusive, final int endExclusive)
```
Comment (yifan-c) [inline cassandra-analytics-common/src/main/java/org/apache/cassandra/spark/utils/RandomUtils.java]: please remove `final`

File: cassandra-analytics-common/src/main/java/org/apache/cassandra/spark/utils/RandomUtils.java [hunk 2]
```diff
@@ -63,6 +64,24 @@ public static int randomPositiveInt(int bound)
...
+        if (startInclusive < 0)
+        {
+            throw new IllegalArgumentException("Both range values must be non-negative.");
+        }
+        if (startInclusive == endExclusive)
+        {
+            return startInclusive;
+        }
+
+        return startInclusive + RANDOM.nextInt(endExclusive - startInclusive);
```
Comment (yifan-c) [inline cassandra-analytics-common/src/main/java/org/apache/cassandra/spark/utils/RandomUtils.java:77]: The bound value for random is exclusive. I think we are missing a value here. 

For example, when `endExclusive - startInclusive == 1`, the produce pseudo random values contains only a single value of 0, but it wants both 0 and 1 instead. 

The util is only used for testing, hence a minor issue. Let's get it correct still.

File: cassandra-analytics-common/src/main/java/org/apache/cassandra/spark/utils/RandomUtils.java [hunk 3]
```diff
@@ -100,8 +119,18 @@ public static InetAddress randomInet()
...
+
+    public static String randomAlphanumeric(int maxLength)
     {
-        return type.randomValue(MIN_COLLECTION_SIZE);
+        StringBuilder sb = new StringBuilder(maxLength);
+        IntStream.rangeClosed(0, maxLength)
+                 .map(i -> RANDOM.nextInt(ALPHANUMERIC.length()))
+                 .mapToObj(ALPHANUMERIC::charAt)
+                 .forEach(sb::append);
+        return sb.toString();
```
Comment (yifan-c) [inline cassandra-analytics-common/src/main/java/org/apache/cassandra/spark/utils/RandomUtils.java]: The alternative in my mind is random from [0, 36), and map the lower 10 to numbers and upper 26 to chars. Therefore, you do not need to look up the ALPHANUMERIC string. 
It is just a nit, feel free to ignore.

File: cassandra-analytics-common/src/main/java/org/apache/cassandra/spark/utils/RandomUtils.java (hunk 2 above)
Comment (jberragan) [reply cassandra-analytics-common/src/main/java/org/apache/cassandra/spark/utils/RandomUtils.java:77]: If it is end exclusive and start inclusive and the diff is 1, then shouldn't it only return 1 value? E.g. (4, 5] should always return 4?

Comment (yifan-c) [reply cassandra-analytics-common/src/main/java/org/apache/cassandra/spark/utils/RandomUtils.java:77]: Ah. Then the if block above should throw IllegalArgumentException too? e.g. [4, 4) is empty 

```java
        if (startInclusive == endExclusive)
        {
            return startInclusive;
        }
```

Comment (jberragan) [reply cassandra-analytics-common/src/main/java/org/apache/cassandra/spark/utils/RandomUtils.java:77]: It currently does this:

```
if (startInclusive == endExclusive)
{
    return startInclusive;
}
```

I don't particularly mind either way, current approach seems more permissive.

**Pull request merged**